addons/account/static/src/js/reconciliation/reconciliation_model.js

```diff
@@ -1126,7 +1126,26 @@ var StatementModel = BasicModel.extend({
         var formatOptions = {
             currency_id: line.st_line.currency_id,
         };
-        var amount = values.amount !== undefined ? values.amount : line.balance.amount;
+        var amount;
+        switch(values.amount_type) {
+            case 'percentage':
+                amount = line.balance.amount * values.amount / 100;
+                break;
+            case 'regex':
+                var matching = line.st_line.name.match(new RegExp(values.amount_from_label_regex))
+                amount = parseFloat(!matching || matching.length != 2 ? null : matching[1].replace(',', '.').replace(/\.(?=.*\.)/, ""));
```

Comment:
I'm not entirely a fan of this as what would happen if the thousand separator is . and we have something like this 100.000 (without any decimals) but I guess that this is a corner case scenario so I can live with it

Reply:
added the decimal separator as a parameter